amplify/backend/api/colonycdapp/schema/schema.graphql

```diff
@@ -1855,6 +1871,24 @@ type User @model {
   """
   liquidationAddresses: [LiquidationAddress!]
     @hasMany(indexName: "byUserAddress", fields: ["id"])
+  """
+  Notifications data for the user
+  """
+  notificationsData: NotificationsData @hasOne(fields: "id")
+}
+
+"""
+Holds the notifications data for the user, such as their unique Magicbell user id, and their notifications preferences.
+"""
+type NotificationsData @model {
+  """
+  Unique identifier for the user
+  """
+  userId: ID!
```

Comment:
I think it's not necessary to add an index to the userId, since it won't be queried by that (as far as I can see atm). I do wonder about changing this schema to instead be like this though:

```graphql
"""
Holds the notifications data for the user, such as their unique Magicbell user id, and their notifications preferences.
"""
type NotificationsData @model {
  """
  Unique identifier for the user
  """
  userId: ID! @primaryKey
  """
  Unique identifier for the user in Magicbell
  """
  magicbellUserId: ID
}
```

Since the id, userId and magicbellUserId are all likely to be the wallet address, so we can streamline it to just use the userId as the primary key (also since this model will _probably_ only ever be queried with the user). I think it's best for now to leave it as it is though, and as this feature develops over time we can revisit this and see if it's still fit for purpose, or if any indexing is needed.

Reply:
All right that makes sense! Cheers @davecreaser

Reply:
@davecreaser how about simply using wallet address as the ID? That way it's deterministic, and you can utilise the auto-generated queries if needed.